src/ucm/cuda/cudamem.c

```diff
@@ -58,6 +58,21 @@
         return ret; \
     }
 
+/* Create a body of CUDA memory release replacement function */
+#define UCM_CUDA_FREE_ASYNC_FUNC(_name, _retval, _ptr_type, _stream_type, \
```

Comment:
can we extend UCM_CUDA_FREE_FUNC to have var-args instead of _stream_type?
see for example how UCM_CUDA_ALLOC_FUNC is using varargs

Reply:
I've been trying to do but failed so far. I tried two different approaches, described below.

The first one would be just to add `_args_format` and `...`, but that would then require existing functions to also pass those empty arguments, and it seems problematic to handle empty var-args, not sure if there are any preprocessor macros currently to handle that case, plus it would pollute the existing `UCM_CUDA_FREE_FUNC` calls with empty parameters.

The second was to pass everything as var-args, including the pointer. We could then extract the first argument (pointer) with `UCS_PP_TUPLE_0`, and then make use of it like the explicit parameter is currently used. However, this would then require passing `__VA_ARGS__` to `ucm_trace` which would need type-casting of the individual variables.

I'm not very experience writing var-arg macros, so maybe you could suggest other approaches?